rclpy/src/rclpy/utils.cpp

```diff
@@ -277,14 +277,117 @@ throw_if_unparsed_ros_args(py::list pyargs, const rcl_arguments_t & rcl_args)
 py::dict
 rclpy_action_get_rmw_qos_profile(const char * rmw_profile)
 {
-  PyObject * pyqos_profile = NULL;
+  py::dict pyqos_profile;
   if (0 == strcmp(rmw_profile, "rcl_action_qos_profile_status_default")) {
-    pyqos_profile = rclpy_common_convert_to_qos_dict(&rcl_action_qos_profile_status_default);
+    pyqos_profile = common_convert_to_qos_dict(&rcl_action_qos_profile_status_default);
   } else {
     std::string error_text = "Requested unknown rmw_qos_profile: ";
     error_text += rmw_profile;
     throw std::runtime_error(error_text);
   }
-  return py::reinterpret_steal<py::dict>(pyqos_profile);
+  return pyqos_profile;
+}
+
+py::object
+_convert_to_py_topic_endpoint_info(const rmw_topic_endpoint_info_t * topic_endpoint_info)
+{
+  py::list py_endpoint_gid = py::list(RMW_GID_STORAGE_SIZE);
+  for (size_t i = 0; i < RMW_GID_STORAGE_SIZE; i++) {
+    py::object py_val_at_index = py::int_(topic_endpoint_info->endpoint_gid[i]);
+    py_endpoint_gid[i] = py_val_at_index;
+  }
+
+  // Create dictionary that represents rmw_topic_endpoint_info_t
+  py::dict py_endpoint_info_dict;
+  // Populate keyword arguments
+  // A success returns 0, and a failure returns -1
+  py_endpoint_info_dict["node_name"] = py::str(topic_endpoint_info->node_name);
+  py_endpoint_info_dict["node_namespace"] = py::str(topic_endpoint_info->node_namespace);
+  py_endpoint_info_dict["topic_type"] = py::str(topic_endpoint_info->topic_type);
+  py_endpoint_info_dict["endpoint_type"] =
+    py::int_(static_cast<int>(topic_endpoint_info->endpoint_type));
+  py_endpoint_info_dict["endpoint_gid"] = py_endpoint_gid;
+  py_endpoint_info_dict["qos_profile"] =
+    common_convert_to_qos_dict(&topic_endpoint_info->qos_profile);
+
+  return py_endpoint_info_dict;
+}
+
+py::list
+convert_to_py_topic_endpoint_info_list(const rmw_topic_endpoint_info_array_t * info_array)
+{
+  if (!info_array) {
+    throw RCLError("rmw_topic_endpoint_info_array_t pointer is empty");
+  }
```

Comment:
Should this check be added to `common_convert_to_qos_dict` and `_convert_to_py_topic_endpoint_info`?